src/components/pod/pod.component.tsx

```diff
@@ -0,0 +1,267 @@
+import React, { useState, useMemo } from "react";
+import { MarginProps } from "styled-system";
+import useLocale from "../../hooks/__internal__/useLocale";
+
+import {
+  StyledBlock,
+  StyledContent,
+  StyledEditAction,
+  StyledActionsContainer,
+  StyledDeleteButton,
+  StyledUndoButton,
+  StyledFooter,
+  StyledPod,
+  StyledHeader,
+  StyledSubtitle,
+  StyledTitle,
+} from "./pod.style";
+import Icon from "../icon";
+
+import Event from "../../__internal__/utils/helpers/events";
+import tagComponent from "../../__internal__/utils/helpers/tags/tags";
+
+export interface PodProps extends MarginProps {
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-component"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-element"?: string;
+  /** Identifier used for testing purposes, applied to the root element of the component. */
+  "data-role"?: string;
+  /** Aligns the title to left, right or center */
+  alignTitle?: "left" | "center" | "right";
+  /** Enables/disables the border around the pod. */
+  border?: boolean;
+  /** Children elements */
+  children?: React.ReactNode;
+  /** Custom className */
+  className?: string;
+  /** Determines the padding around the pod */
+  size?: "extra-small" | "small" | "medium" | "large" | "extra-large";
+  /** Prop to apply a theme to the Pod */
+  variant?: "primary" | "secondary" | "tertiary" | "tile" | "transparent";
+  /** Title for the pod h4 element always shown */
+  title?: string | React.ReactNode;
+  /** Optional subtitle for the pod */
+  subtitle?: string;
+  /** A component to render as a Pod footer */
+  footer?: string | React.ReactNode;
+  /** Supplies an edit action to the pod */
+  onEdit?:
+    | string
+    | Record<string, unknown>
```

Comment:
Looking at the code I assume we previously allowed consumers to specify an external link as a string/object for `onEdit`, which may explain why the button itself is an `a` tag. Since the actual code does not work:
```typescript
// previously in pod.component.js
function linkProps() {
      if (typeof onEdit === "string") {
        return { to: onEdit }; // the `to` prop for `Link` was removed a long time ago
      }

      return onEdit;
    }
```
and we have not had any issues raised about this, I would assume very few projects would be using this pattern. But I will speak with UX if they are happy for me to remove this and make the edit button a `button`

Reply:
After discussing this pattern with @harpalsingh, he agrees there should not be a problem with removing it 👍🏼

Reply:
Raised FE-5369 for this

Reply:
Nice one 👍